contracts/src/v0.8/llo-feeds/v0.5.0/configuration/ChannelConfigStore.sol

```diff
@@ -2,24 +2,121 @@
 pragma solidity ^0.8.19;
 
 import {ConfirmedOwner} from "../../../shared/access/ConfirmedOwner.sol";
+import {EnumerableSet} from "@openzeppelin/contracts@4.8.3/utils/structs/EnumerableSet.sol";
 
 import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
 import {IChannelConfigStore} from "./interfaces/IChannelConfigStore.sol";
 
 contract ChannelConfigStore is ConfirmedOwner, IChannelConfigStore, ITypeAndVersion {
+  using EnumerableSet for EnumerableSet.UintSet;
+
   event NewChannelDefinition(uint256 indexed donId, uint32 version, string url, bytes32 sha);
+  event ChannelDefinitionAdded(uint256 indexed donId, uint32 indexed channelAdderId, string url, bytes32 sha);
+  event ChannelAdderSet(uint256 indexed donId, uint32 indexed channelAdderId, bool allowed);
+  event ChannelAdderAddressSet(uint32 indexed channelAdderId, address adderAddress);
+
+  error UnauthorizedChannelAdder();
 
   constructor() ConfirmedOwner(msg.sender) {}
 
   /// @notice The version of a channel definition keyed by DON ID
   // Increments by 1 on every update
   mapping(uint256 => uint256) internal s_channelDefinitionVersions;
 
-  function setChannelDefinitions(uint32 donId, string calldata url, bytes32 sha) external onlyOwner {
+  /// @notice Mapping from channel adder ID to its corresponding address
+  mapping(uint32 => address) internal s_channelAdderAddresses;
+
+  /// @notice Mapping from DON ID to the set of allowed channel adder IDs
+  mapping(uint256 => EnumerableSet.UintSet) internal s_allowedChannelAdders;
+
+  function setChannelDefinitions(
+    uint32 donId,
+    string calldata url,
+    bytes32 sha
+  ) external onlyOwner {
     uint32 newVersion = uint32(++s_channelDefinitionVersions[uint256(donId)]);
     emit NewChannelDefinition(donId, newVersion, url, sha);
   }
 
+  /// @notice Allows a channel adder to add channel definitions to the specified DON.
+  /// The DON enforces (in its consensus rules), that the channel definitions provided
+  /// by the channel adder are well-formed, purely additive, and do not overlaod the DON.
+  /// @param donId The DON ID
+  /// @param channelAdderId The channel adder ID
+  /// @param url The URL of the channel definition
+  /// @param sha The SHA hash of the channel definition
+  function addChannelDefinitions(
```

Comment:
Should this function check that a `s_channelDefinitionVersions` exists for particular `donId` i.e. we've called `setChannelDefinitions` at least once?